in_toto/gpg/util.py

```diff
@@ -247,3 +310,37 @@ def is_version_fully_supported():
 
   else: # pragma: no cover
     return False
+
+
+def get_hashing_class(hash_algorithm_id):
+  """
+  <Purpose>
+    Return a pyca/cryptography hashing class reference for the passed RFC4880
+    hash algorithm ID.
+
+  <Arguments>
+    hash_algorithm_id:
+            one of SHA1, SHA256, SHA512 (see in_toto.gpg.constants)
+
+  <Exceptions>
+    ValueError
+            if the passed hash_algorithm_id is not supported.
+
+  <Returns>
+    A pyca/cryptography hashing class
+
+  """
+  if hash_algorithm_id == in_toto.gpg.constants.SHA1:
+    return hashing.SHA1
+
+  elif hash_algorithm_id == in_toto.gpg.constants.SHA256:
+    return hashing.SHA256
+
+  elif hash_algorithm_id == in_toto.gpg.constants.SHA512:
+    return hashing.SHA512
+
+  else:
+    raise ValueError("Hash algorithm '{}' not supported, must be one of '{}' "
+        "(see RFC4880 9.4. Hash Algorithms).".format(hash_algorithm_id,
+        {in_toto.gpg.constants.SHA1, in_toto.gpg.constants.SHA256,
+         in_toto.gpg.constants.SHA512}))
```

Comment:
If this is never extended to support other hash types, my comment is moot, but there's a perhaps more concise way that avoids the `if` cascade:
```python
  supported_hashing_algorithms = [in_toto.gpg.constants.SHA1,
      in_toto.gpg.constants.SHA256, in_toto.gpg.constants.SHA512]

  corresponding_hashing_classes = [hashing.SHA1, hashing.SHA256, hashing.SHA512]

  hashing_class = dict(zip(supported_hashing_algorithms, corresponding_hashing_classes))

  try:
    return hashing_class[hash_algorihm_id]
  except KeyError:
    raise ValueError("Hash algorithm '{}' not supported, must be one of '{}' "
        "(see RFC4880 9.4. Hash Algorithms).".format(hash_algorithm_id,
        supported_hashing_algorithms))
```

(And had 938856026d091878883db9f837fe762dceaa4c03 not dropped `SUPPORTED_HASH_ALGORITHMS` from `gpg/constants.py`, the proposed code could be even DRYer.)

Reply:
I opted for the if/else variant because it seemed a tad easier to read, and I am not sure if we need to support other algos in the near future. But your approach is definitely more elegant.